### PR TITLE
Rename `BlockContext` to `ModelContext`

### PR DESCRIPTION
**Pull request by karlhigley**
Peeling off a small piece from the API revisions in #368, this is a straightforward rename without any changes in behavior.

### CODE REVIEW

File: merlin/models/tf/__init__.py [hunk 1]
```diff
@@ -108,8 +104,13 @@
 from merlin.models.tf.prediction_tasks.next_item import NextItemPredictionTask
 from merlin.models.tf.prediction_tasks.regression import RegressionTask
 from merlin.models.tf.prediction_tasks.retrieval import ItemRetrievalTask
+
+# Must happen before any importing of tensorflow to curtail mem usage
 from merlin.models.tf.utils import repr_utils
 
+configure_tensorflow()
```
Comment (marcromeyn) [inline merlin/models/tf/__init__.py:111]: Why did this move? Isn’t this after the tensorflow imports now?

Comment (karlhigley) [reply merlin/models/tf/__init__.py:111]: ugh isort

Comment (karlhigley) [reply merlin/models/tf/__init__.py:111]: Something about the merge commit is...weird

Comment (karlhigley) [reply merlin/models/tf/__init__.py]: Judging by the stray comment down at the bottom I had to remove, seems like I'm not the first person this has happened to. 😅

**Review by marcromeyn — APPROVED** [review]

**Pull request merged**